db/db_impl/db_impl.cc

```diff
@@ -3720,6 +3721,28 @@ ArenaWrappedDBIter* DBImpl::NewIteratorImpl(
   return db_iter;
 }
 
+std::unique_ptr<MultiCfIterator> DBImpl::NewMultiCfIterator(
+    const ReadOptions& _read_options,
+    const std::vector<ColumnFamilyHandle*>& column_families) {
+  assert(column_families.size() > 0);
```

Comment:
In public APIs, at least if a precondition is easy to check, I believe we prefer to check it even in release builds and return an appropriate error (rather than crashing).

Reply:
(Complex preconditions like "these keys are sorted" would cost substantially in performance to check)